### ISSUE TITLE
Example R / Scala notebooks

### ISSUE THREAD

**Issue by parente** [jupyter-incubator/dashboards_server#203]
Need some notebooks in `test/` for manually testing that R and Scala both work properly.

* ~~[ ] R notebook that just prints~~ See comment about R support
* ~~[ ] Scala notebook that just prints~~(Should go in #211 for simple manual testing)
* ~~[ ] R notebook that uses minimal declarative widgets~~ See comment about R support
* [x] Scala notebook that uses minimal declarative widgets (This is now the one and only goal)

Don't want to add massive new demos that we need to maintain. Just want a sanity check of how these things behave in dashboard server.

Needs to come after #169

**Comment by parente** [comment]
https://github.com/IRkernel/IRkernel/pull/312 fixes the R kernel so that it works with jupyter-js-services. Not sure when there will be a release with it, so for the time being, focus on Scala and Toree.

**Comment by parente** [comment]
Description updated. Moved this out of 0.6.0.

**Comment by parente** [comment]
We did not check / install the necessary Toree declarative widget deps in #169. They might already be installed properly in the Dockerfile.kernel or they might not. If not, that work will need to be done as part of this issue.

**Comment by jameslmartin** [comment]
We have a choice to make concerning the declarative widgets .jar file for Scala. Currently, we ask the author to specify the IP and port of the notebook server to successfully load the .jar file. 
```scala
// modify to IP and Port of this notebook server
%addjar http://localhost:8888/nbextensions/declarativewidgets/declarativewidgets.jar

import declarativewidgets._

initWidgets
```
This introduces a problem when deploying the notebook to a dashboard server, as the dashboard server should stand alone and does not rely on a notebook server.

@parente made one suggestion that we could move the .jar file to be public facing, but we may then run into problems with versioning. We could also load the .jar file using the path to where the file is in the local file system of the Docker notebook server container. This doesn't quite solve the portability issue unless we put the .jar file along the same path in the dashboard server.

Any other suggestions or workarounds?

**Comment by parente** [comment]
@lbustelo, @jhpedemonte any thoughts on how the declarativewidgets.jar should be referenced kernel-side so that it works both in notebook server and in kernel gateway (deployed)?

An analogous `import foo` in Python works because we assume the notebook environment and the KG environment both have package `foo`. The `addjar` is more complicated because an absolute file or URL has to be given. (Unless it searches the classpath too?)

**Comment by jameslmartin** [comment]
Spoke with @jhpedemonte, @lbustelo, and @poplav about this. We decided that publishing tagged builds of declarative widgets to Maven Central may be best and easiest for users. This will require a change to how widgets are initialized in Scala notebooks, but will also solve the issue of versioning.

We're opening a separate issue on declarative widgets, which I'll link here when posted.

**Comment by jameslmartin** [comment]
Thanks to @poplav's work on https://github.com/jupyter-incubator/declarativewidgets/pull/468, declarative widgets for Scala is now hosted at Maven Central. I've created an example Scala notebook that pulls down the current snapshot release and submitted PR #267.
